libraries/rush-lib/src/logic/operations/OperationExecutionRecord.ts

```diff
@@ -166,6 +166,17 @@ export class OperationExecutionRecord implements IOperationRunnerContext {
     return this._operationMetadataManager?.stateFile.state?.cobuildRunnerId;
   }
 
+  public get isTerminal(): boolean {
+    return TERMINAL_STATUSES.has(this.status);
+  }
+
+  public get isRemoteExecuting(): boolean {
+    return (
+      this.status === OperationStatus.RemoteExecuting ||
```

Comment:
Status change to `RemoteExecuting` doesn't make new work available on the queue. We only need to kick `assignOperations()` if the status change caused an operation to go from unavailable for execution -> available.